### DISCUSSION TITLE
Google fonts: Use individual ttfs or "VariableFont"?

### DISCUSSION THREAD

**Discussion by GregDomzalski** [AvaloniaUI/Avalonia#16355] [Q&A] [answered]
Hello. I'm using custom fonts from Google Fonts as suggested in the documentation. When I look at the archive I'm downloading, there are two sets of files: 1) The individual ttfs inside of the `static` folder for each of the supported font weights, and 2) a single `-VariableFonts_wght.ttf` file in the root of the archive.

Does Avalonia support consuming these `-VariableFonts_wght.ttf` files? I attempted to use it in the same manner as the other fonts and I got the standard `Could not create glyphTypeface` exception message that I see when Avalonia can't find the font.

It would be nice to only have to copy one or two files instead of a dozen or so files for each font family. Not the biggest deal, but I figured I'd ask if this is supported or if I'm doing anything wrong here. 😄

Font is located at: `/Assets/Fonts/RobotoMono-VariableFont_wght.ttf`

Relevant csproj section:

```xml
  <ItemGroup>
    <AvaloniaResource Include="Assets\**" />
  </ItemGroup>
```

App.xaml:

```xml
<Application xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             x:Class="TestApp.App"
             xmlns:local="using:TestApp"
             RequestedThemeVariant="Default">
    <!-- "Default" ThemeVariant follows system theme variant. "Dark" or "Light" are other available options. -->
    <Application.DataTemplates>
        <local:ViewLocator />
    </Application.DataTemplates>

    <Application.Styles>
        <FluentTheme />
    </Application.Styles>

    <Application.Resources>
        <!-- Other fonts -->
        <FontFamily x:Key="RobotoMonoFont">avares://TestApp/Assets/Fonts#RobotoMono</FontFamily>
    </Application.Resources>
</Application>
```

Usage:

```xml
<TextBlock
    Text="Test message"
    FontFamily="{StaticResource RobotoMonoFont}"
    FontWeight="Bold" />
```

Please let me know if there's additional information that would be useful to share.

**Comment by Gillibald** [comment]
The font is most likely called "Roboto Mono"

**Reply by GregDomzalski** [reply]
Ah 🤦🏻‍♂️ OK - that got the font loading. Is it correct to assume that the name used after the `#` should match what my font viewer utility calls it?

**Reply by Trevo525** [reply]
> Ah 🤦🏻‍♂️ OK - that got the font loading. Is it correct to assume that the name used after the `#` should match what my font viewer utility calls it?

So to be clear, the part that fixed your problem was changing:
`<FontFamily x:Key="RobotoMonoFont">avares://TestApp/Assets/Fonts#RobotoMono</FontFamily>`

to 

`<FontFamily x:Key="RobotoMonoFont">avares://TestApp/Assets/Fonts#Roboto Mono</FontFamily>`

**Comment by GregDomzalski** [comment]
OK - so the font is now loading, thanks @Gillibald!

So now back to the other part of the question: Does Avalonia support these variable weight fonts?

The font is now rendering correctly, but setting `FontWeight` no longer seems to have an effect. That would seem to indicate it is not supported - but given how new I am to Avalonia and XAML frameworks, I would not rule out user error. I had done a search for "VariableFonts" in both issues and discussions but did not see anything. I may also just be searching on the wrong terms?

**Reply by Gillibald** [reply, marked as answer]
Variable font support is not yet implemented but will be later this year

**Reply by GregDomzalski** [reply]
Thanks! I really appreciate the replies.

**Reply by LaurentInSeattle** [reply]
Variable fonts do work for me...
I am using Inter-V and I activated it using this code below: 

`
class Program
{
    [STAThread]
    public static void Main(string[] args) => BuildAvaloniaApp() .StartWithClassicDesktopLifetime(args);

    public static AppBuilder BuildAvaloniaApp()
        => AppBuilder.Configure<App>().UsePlatformDetect().WithInterVariableFont().LogToTrace();
}

public static class AppBuilderExtension
{
    public static AppBuilder WithInterVariableFont(this AppBuilder appBuilder)
        => appBuilder.ConfigureFonts(fontManager =>
        {
            fontManager.AddFontCollection(new InterVariableFontCollection());
        });    
}

public sealed class InterVariableFontCollection : EmbeddedFontCollection
{
    public InterVariableFontCollection() : base(
        new Uri("fonts:Inter-V", UriKind.Absolute),
        new Uri("avares://MyApp/Assets/Fonts/Inter-V.ttf", UriKind.Absolute)) {  }
}

`

Is this something that's going to break in future versions ? Or am I missing something? 
Thanks

**Reply by Gillibald** [reply]
You can't modify any variable axis. So it is not supported.

**Reply by LaurentInSeattle** [reply]
Well... The above code allows me to correctly display text in Italics in various TextBlock's in my test app'.
Italics font files for Inter are not shipping with Avalonia 11.0.11, if I understand correctly...
Not sure about what's going on here. Italics are special?

**Reply by Gillibald** [reply]
I have added some way to emulate italics if that variant isn't available in the font collection but a regular one

**Reply by Trevo525** [reply]
> Variable font support is not yet implemented but will be later this year

Did that ever get implemented? I don't see a PR closed or otherwise with "Variable Fonts" in the title.

**Reply by Gillibald** [reply]
This feature is still planned. There was not enough time to work on this last year.
